### PR TITLE
Some typing improvements, add py.typed

### PR DESCRIPTION
**Pull request by micimize**
# Description

* Improves some method typing for the `MinigridEnv` api and elsewhere.
* Adds a `py.typed` marker so that tooling like mypy will know to use minigrid's type annotations.
* Changed name in `setup.py` to `minigrid`

I did not find a `CONTRIBUTING.md` but did `pre-commit` and ensured all tests pass.

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# Checklist:

- [X] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files`
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def get_tests_requirements():
 header_count, long_description = get_description()
 
 setup(
-    name="Minigrid",
+    name="minigrid",
```
Comment (pseudo-rnd-thoughts) [inline setup.py:54]: Does setuptool just make everything lower case?

**Review by pseudo-rnd-thoughts — APPROVED** [review]
Thanks for the PR, this is very helpful. 

Do you use minigrid a lot? Would you be interested in helping with more research?

File: setup.py (hunk 1 above)
Comment (micimize) [reply setup.py:56]: `pip install -e` will install with this name, case sensitive – I'm not sure why this was working for `setuptools`

**Comment by micimize** [comment]
@pseudo-rnd-thoughts ok – I've added an `--add-import` to `isort` along with `autoflake` to remove the leftover imports, and changed the ordering to reduce the need for re-running `pre-commit` (`pyupgrade` removes need for imports, removing imports effects isort ordering)

I have only really just started using it for my project – some details here: https://github.com/Farama-Foundation/Minigrid/issues/168#issuecomment-1307849050. But would be happy to join the discord – can you resend an invite? It became invalid when discord tried to make me a new account :/

File: setup.py (hunk 1 above)
Comment (pseudo-rnd-thoughts) [reply setup.py:56]: If it is case sensitive, could you revert this change

**Comment by pseudo-rnd-thoughts** [comment]
Thanks for the changes, here is a new link https://discord.gg/N8CwYyGt
Could you revert the setup name then we can merge

**Review by pseudo-rnd-thoughts — APPROVED** [review]

**Pull request merged**